core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java

```diff
@@ -277,13 +279,18 @@ private void fillInstanceMetadata(Stack stack) {
         }
     }
 
-    private boolean shouldUseBaseImage(ClusterV4Request clusterRequest, Blueprint blueprint) {
+    private boolean shouldUseBaseAmbariImage(ClusterV4Request clusterRequest) {
         return (clusterRequest.getAmbari() != null && clusterRequest.getAmbari().getRepository() != null)
                 || (clusterRequest.getAmbari() != null
                 && clusterRequest.getAmbari().getStackRepository() != null
                 && clusterRequest.getAmbari().getStackRepository().customRepoSpecified());
     }
 
+    boolean shouldUseBaseCMImage(ClusterV4Request clusterRequest) {
+        ClouderaManagerV4Request cmRequest = clusterRequest.getCm();
+        return (cmRequest != null && !isEmpty(cmRequest.getProducts())) || (cmRequest != null && cmRequest.getRepository() != null);
```

Comment:
This isEmpty() is for strings not lists and this conditional could be expressed in a simpler form